lib/rules/no-error-cause.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @author Sosuke Suzuki <https://github.com/sosukesuzuki>
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict"
+
+const { CONSTRUCT, READ, ReferenceTracker } = require("eslint-utils")
+
+/**
+ * @typedef {import("estree")} Node
+ * @typedef {import("estree").ClassExpression | import("estree").ClassDeclaration} ClassNode
+ * @typedef {import("estree").CallExpression} CallExpression
+ */
+
+const traceMap = {}
+for (const errorConstructorName of ["Error", "AggregateError"]) {
+    traceMap[errorConstructorName] = { [CONSTRUCT]: true, [READ]: true }
+}
+
+/**
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function isSuperCall(node) {
+    return node.type === "CallExpression" && node.callee.type === "Super"
+}
+
+/**
+ * Checks if the received node is a construcotr call with cause option.
+ * e.g. `new Error("message", { cause: foo })`, `super("message", { cause: foo })`
+ *
+ * @param {Node} node
+ * @returns {boolean}
+ */
+function isConstructCallWithCauseOption(node) {
+    if (node.type !== "NewExpression" && !isSuperCall(node)) {
+        return false
+    }
+    const secondArg = node.arguments[1]
+    if (!secondArg || secondArg.type !== "ObjectExpression") {
+        return false
+    }
+    return secondArg.properties.some((property) => {
+        if (property.type !== "Property") {
+            return false
+        }
+        // new Error("msg", { ["cause"]: foo })
+        if (
+            property.computed &&
+            property.key.type === "Literal" &&
+            property.key.value === "cause"
+        ) {
+            return true
+        }
+        // new Error("msg", { cause: foo })
+        return property.key.name === "cause"
```

Comment:
Should check `.computed` here too, `new Error(...foo, {[cause]: ''})`

Reply:
I don't think we should check `.computed` here.

```js
const cause = "not-cause";
new Error("message", { [cause]: '' });
```

Reply:
I don't understand, why not? The code above is a false positive, isn't it?